src/js/components/ToggleButtonGroup/__tests__/ToggleButtonGroup-test.tsx

```diff
@@ -0,0 +1,490 @@
+    await user.type(toggleButtonOne, '{arrowDown}');
+    await user.type(toggleButtonTwo, '{arrowRight}');
+    await user.type(toggleButtonTwo, '{enter}');
+    expect(toggleButtonTwo).toHaveAttribute('aria-checked', 'true');
+  });
+
+  test('Should move to previous button left/up arrow key is pressed and loop', async () => {
+    const user = userEvent.setup();
+    render(
+      <Grommet>
+        <ToggleButtonGroup options={['one', 'two', 'three']} />
+      </Grommet>,
+    );
+
+    const toggleButtonOne = screen.getByRole('radio', {
+      name: /one/i,
+    });
+    const toggleButtonTwo = screen.getByRole('radio', {
+      name: /two/i,
+    });
+
+    await user.tab();
+    await user.type(toggleButtonOne, '{arrowLeft}');
+    await user.type(toggleButtonTwo, '{arrowUp}');
+    await user.type(toggleButtonTwo, '{enter}');
+    expect(toggleButtonTwo).toHaveAttribute('aria-checked', 'true');
+  });
+
+  test('icon with values', () => {
+    const { container } = render(
+      <Grommet>
+        <ToggleButtonGroup
+          options={[
+            { icon: <Add />, value: 'one' },
+            { icon: <Add />, value: 'two' },
+          ]}
+          defaultValue="one"
+        />
+      </Grommet>,
+    );
+
+    expect(container).toMatchSnapshot();
+  });
+
+  test('string options with multiple prop', () => {
+    const { container } = render(
+      <Grommet>
+        <ToggleButtonGroup multiple options={['one', 'two']} />
+      </Grommet>,
+    );
+
+    expect(container.firstChild).toMatchSnapshot();
+  });
+
+  test('Should render when options is array of objects with multiple prop', () => {
+    const { container } = render(
+      <Grommet>
+        <ToggleButtonGroup
+          multiple
+          options={[
+            { id: 'onE', label: 'One', value: 'one' },
+            { id: 'twO', label: 'Two', value: 'two' },
+          ]}
+        />
+      </Grommet>,
+    );
+
+    expect(container.firstChild).toMatchSnapshot();
+  });
+
+  test('Should call onChange in uncontrolled scenarios with multiple prop', () => {
+    const onChange = jest.fn();
+    const { container } = render(
+      <Grommet>
+        <ToggleButtonGroup
+          options={['one', 'two', 'three']}
+          onChange={onChange}
+          multiple
+        />
+      </Grommet>,
+    );
+    expect(container.firstChild).toMatchSnapshot();
+
+    // Simulate a click event on the buttons
+    fireEvent.click(screen.getByRole('button', { name: 'two' }));
+    fireEvent.click(screen.getByRole('button', { name: 'three' }));
+    expect(onChange).toHaveBeenCalledWith(['two', 'three']);
+  });
+
+  test('Should render defaultValue as active with multiple prop', () => {
+    render(
+      <Grommet>
+        <ToggleButtonGroup
+          options={[
+            { label: 'Choice 1', value: 'c1' },
+            { label: 'Choice 2', value: 'c2' },
+            { label: 'Choice 3', value: 'c3' },
+          ]}
+          multiple
+          defaultValue="c2"
```

Comment:
In `multiple` scenarios, value/defaultValue should be an array of strings

Reply:
For this, can we also have multiple values as default value --> ['c2', 'c3']

Reply:
done